### PR TITLE
implement thrad-safe singleton pattern based on c++11

### PR DESCRIPTION
**Pull request by jovany-wang**
use local static var to archive a thread safety singleton pattern.

### CODE REVIEW

File: SingletonPattern/SingletonPattern.cpp [hunk 1]
```diff
@@ -3,25 +3,30 @@ using namespace std;
...
-Singleton* Singleton::_instance = nullptr;
+
 
 int main() {
-    Singleton *singleton1 = Singleton::Instance();
-    cout << singleton1 << endl;
-    Singleton *singleton2 = Singleton::Instance();
-    cout << singleton2 << endl;
+    Singleton &singleton1 = Singleton::Instance();
+    singleton1.printInstanceAddr();
```
Comment (micooz) [inline SingletonPattern/SingletonPattern.cpp]: `printInstanceAddr()` is wordy, why not `cout << &singleton1 << endl;` directly?

File: SingletonPattern/SingletonPattern.cpp [hunk 2]
```diff
@@ -3,25 +3,30 @@ using namespace std;
...
+
+    void printInstanceAddr() {
+        cout << this << endl;
+    }
+
 private:
-    Singleton() { cout << "Singleton..." << endl; }
-    static Singleton* _instance;
+    Singleton() {
+        cout << "ctor" << endl;
```
Comment (micooz) [inline SingletonPattern/SingletonPattern.cpp]: What does `ctor` mean here?

**Review by micooz — CHANGES_REQUESTED** [review]

File: SingletonPattern/SingletonPattern.cpp (hunk 2 above)
Comment (jovany-wang) [reply SingletonPattern/SingletonPattern.cpp]: It shows how many times the `ctor`  has been called.

File: SingletonPattern/SingletonPattern.cpp (hunk 1 above)
Comment (jovany-wang) [reply SingletonPattern/SingletonPattern.cpp]: Silly me.
I will rewrite it.

**Pull request merged**